Exclude pages where `shouldRegisterNavigation()` is false

oddvalue opened the pull request:
Fixes #33

pxlrbt commented:
Hey,

thanks for your PR and sorry for the delayed response. Since pages with `shouldRegisterNavigation()` might still be accessible, I'd like to keep it this way. We could introduce a method `shouldExcludeFromSpotlight()` though and check whether that method exists before registering.

What do you think?

oddvalue commented:
@pxlrbt yep, that makes sense. I've updated the PR and added a section to the readme.

It is worth noting, though, that this solution won't solve the issue reported in #17 or any other vendor page that requires a url parameter.

pxlrbt commented:
Thanks!
